remote-week-2: add Express

YuakaTL opened the pull request:
Elastic IP: 35.75.125.109
DB Host: appworks.ciq6zrrs86t1.ap-northeast-1.rds.amazonaws.com
Changes:
- Add Express and implement the MVC architecture.
- Set up PM2 and Nginx on EC2 instance for better process management and server handling. ([reference](https://medium.com/@nishankjaintdk/setting-up-a-node-js-app-on-a-linux-ami-on-an-aws-ec2-instance-with-nginx-59cbc1bcc68c))
- Change EC2 timezone to Asia/Taipei
- Add content to the README file.

## Synchronous vs. Asynchronous Function
![SCR-20230224-x3q](https://user-images.githubusercontent.com/69000806/221223840-7a76b84c-0435-4221-ba96-f8da90c8974d.png)

- Based on the result of experiment, what's the differences of sync/async functions we can tell?
可以發現同步函數在執行過程中會阻塞線程，直到函數執行完畢，因此在處理大量數據或需要長時間運行的操作時，同步函數的性能可能會比較差，而非同步函數在執行過程中不會阻塞線程，

- In which types of situations are asynchronous functions most useful?
  I/O bound requests.
Examples : writing/reading to a file or database, making API calls, calling hardware like printers etc.

- We've known the 3 ways to implement async functions, how do we choose from them and what are their pros and cons?
  - Callback Function
   > pros
    可以自己定義函式執行的順序

   > cons
    非同步 Callback 很容易發生 [Callback Hell](http://callbackhell.com/)
  - Promise 
  > pros
  解決了 Callback Hell 的問題，將非同步操作以同步操作的流程表達出來

  > cons
  Promise 一旦被創建，就無法取消，可能會導致資源浪費
  - Async/Await
  > pros
  處理複雜流程時，程式較易讀
  
  >cons
  無法處理 promise 返回的 reject 對象，要藉助 try catch

YuakaTL committed:
feat: 🎸 finish remote-assignment-week-2

YuakaTL committed:
docs: ✏️ update readme

YuakaTL committed:
test: 💍 add thunder-collection

addie-tyc commented:
>   2. CPU bound requests (requires CPU time).
>      Examples : looping through thousands of objects, complex lengthy calculations, processing millions of data points etc

Single thread 的非同步真的有適合 CPU bound 的場景嗎？

addie-tyc commented on controllers/userController.js:
```diff
@@ -0,0 +1,62 @@
+            if (err) return console.log(err);
+            let data = {
+                user: results[0],
+                date: new Date().toUTCString()
```
題目要求是要回傳 headers 裡的 `Request-Date` > <

addie-tyc replied:
題目要求是要回傳 headers 裡的 Request-Date > <

addie-tyc commented on controllers/userController.js:
```diff
@@ -0,0 +1,62 @@
+                        }
+                        successHandle(res, "新增 User 成功", data);
+                    })
+                })
```
這個 password 應該什麼都拿不到？XD

addie-tyc reviewed (approved):
只有跟需求小小不符合，調整完可以直接 merge，專案切分和架構很不錯，現階段和下階段的方方面面都顧到了，~~請當小老師 > <~~
~~是個喜歡 callback 的孩子呢~~

YuakaTL committed:
refactor: 💡 Update MySQL connection to async/await

Refactor the code to use a Pool Promise connection with MySQL2 for
improved database connection management. This commit also updates the
userModel and userController files to utilize async/await syntax for
improved readability and to take advantage of the benefits of
asynchronous programming.

addie-tyc reviewed (approved):
Well done. 真是不好意思還讓你跑去重構 QQ
